Created RouterModule:

maxzag opened the pull request:
- created new pages (auth, library, video)
- router types
- router context/provider
- router service (with navigate method)

Commit:
Created RouterModule:
- created new pages (auth, library, video)
- router types
- router context/provider
- router service (with navigate method)

Rastopyr commented on src/Router/index.ts:
```diff
@@ -0,0 +1,2 @@
+export * from './types';
```
i would like suggest to avoid index.ts files with only exports from another file. it increase count of links to this files, and increase coupling. im not sure about helpful of this type of file

maxzag replied:
I used to clean imports. 
From: 

```
import { FC } from 'react';
import { Routes, Route, BrowserRouter } from 'react-router-dom';
import { RouterProvider } from "../../Router/RouterContext";
import { RouteName } from "../../Router/types";
import { Library } from "../../pages/library/library";
import { Login } from "../../pages/auth/login/login";
import { Registration } from "../../pages/auth/registration/registration";
import { Video } from "../../pages/library/video/video";
```

to

```
import { FC } from 'react';
import { Routes, Route, BrowserRouter } from 'react-router-dom';
import { Library, Login, Registration, Video } from '../../pages';
import { RouteName, RouterProvider } from '../../Router';
```

Rastopyr replied:
Yes, i understand. But first approach is more proffered. Code is cleaner, but code is much more coupled

Rastopyr commented on src/Router/types.ts:
```diff
@@ -0,0 +1,24 @@
+    navigate: (path: string) => void;
+}
+
+export enum RouteName {
```
Value of enum keys should match with values.

If you need to define routes, i would suggest to do to in implementation.

```ts
function getRoutePath(route: RouteName): string {
  switch(route) {
    case RouteName.LOGIN:
       return "/login";
    // ...
    default:
      return assertInvalidValue(route);
  }
}
```

maxzag replied:
@Rastopyr 
But what I should do in case I need to put query or params? like video/:video_slag

Rastopyr replied:
Not sure, that i understand the problem. Can you pls provide some scenario, where problem is highlighted?

Rastopyr commented on src/Router/types.ts:
```diff
@@ -0,0 +1,24 @@
+    VIDEO = '/video'
+}
+
+export type RouterContextType = {
```
There is a problem of type definition in this fille. It explain type for 2 levels. O would llike suggest define types forr React context near of this context definitioon

maxzag replied:
@Rastopyr 
So better to don't create the `types.js` files? 
Better to declare types in related files? Like `IRouter` in RouterSerives file and `RouterContextType` in RouterContext?

Rastopyr replied:
If you can to not create it - avoid creation of `types.ts`. In some scenarios we will need it, but not in this case.

Commit:
Made some fixes:
- remove all index.ts files, to increase coupling
- create getRoutePath function to fix problem with enum definition
- remove type.ts file from Router module, because it wasn't necessary

Rastopyr commented on src/Router/RouterService.ts:
```diff
@@ -0,0 +1,19 @@
+    private readonly navigator: RouterNavigateFunction;
+
+    constructor(navigator: RouterNavigateFunction) {
+        this.navigator = navigator;
```
`navigator` -> `navigate`

Rastopyr reviewed (approved):
LGTM, but left one comment

Commit:
Rename router service variable from navigator to navigate
